opentelemetry-sdk/src/metrics/periodic_reader.rs

```diff
@@ -352,23 +324,18 @@ impl PeriodicReaderInner {
         }
     }
 
-    fn collect_and_export(&self, timeout: Duration) -> MetricResult<()> {
+    fn collect_and_export(&self) -> MetricResult<()> {
         // TODO: Reuse the internal vectors. Or refactor to avoid needing any
         // owned data structures to be passed to exporters.
         let mut rm = ResourceMetrics {
             resource: Resource::empty(),
             scope_metrics: Vec::new(),
         };
 
-        // Measure time taken for collect, and subtract it from the timeout.
         let current_time = Instant::now();
         let collect_result = self.collect(&mut rm);
         let time_taken_for_collect = current_time.elapsed();
-        let _timeout = if time_taken_for_collect > timeout {
-            Duration::from_secs(0)
-        } else {
-            timeout - time_taken_for_collect
-        };
+
```

Comment:
current_time, time_taken can also be removed now.

Reply:
This is being used for otel_debug logging. Do you want to remove that?

Reply:
oh got it. We can keep it.